### PR TITLE
feat: deepen shell command projection seam

### PR DESCRIPTION
**Pull request by trek-e**
## Feature PR

> **Using the wrong template?**
> — Bug fix: use [fix.md](?template=fix.md)
> — Enhancement to existing behavior: use [enhancement.md](?template=enhancement.md)

---

## Linked Issue

> **Required.** This PR will be auto-closed if no valid issue link is found.
> The linked issue **must** have the `approved-feature` label. If it does not, this PR will be closed without review — no exceptions.

Closes #3439

> ⛔ **No `approved-feature` label on the issue = immediate close.**
> Do not open this PR if a maintainer has not yet approved the feature spec.
> Do not open this PR if you wrote code before the issue was approved.

---

## Feature summary

Implements a first migration slice of the Shell Command Projection Module by centralizing command projection policy used by managed hook rewrites, local hook command construction, Windows shim text projection, and SDK PATH diagnostic projection.

### New files

| File | Purpose |
|------|---------|
| N/A | No new files in this slice |

### Modified files

| File | What changed |
|------|-------------|
| `get-shit-done/bin/lib/shell-command-projection.cjs` | Added shell command projector, managed-hook surface policy, legacy settings rewrite projector, Windows shim triple builder, and SDK path diagnostic projection helpers |
| `bin/install.js` | Delegated hook rewrite projection, local hook command projection, Windows shim triple projection, and SDK path diagnostic projection to the module seam |
| `tests/bug-3413-shell-command-projection.test.cjs` | Added regression coverage for new projection seam helpers |
| `CLAUDE.md` | Updated repo GitHub auth guidance text present in working tree for this branch |
| `CONTEXT.md` | Added AI Ops memory entries present in working tree for this branch |
| `docs/agents/issue-tracker.md` | Updated gh auth guidance text present in working tree for this branch |
| `docs/installer-migrations.md` | Clarified preserve-user behavior and Antigravity source-note language present in working tree for this branch |

## Implementation notes

- This PR intentionally focuses on a migration slice of #3439 rather than the full issue scope.
- Managed hook allowlist policy is now module-owned and surface-aware (`settings-json` vs `codex-toml`).
- Local `.sh` hook command construction now uses the same projection seam and runner resolution path as other hook surfaces.

## Spec compliance

- [x] GSD has one named module that owns projection from typed command intent to shell/runtime-specific command text.
- [x] `bin/install.js` no longer owns ad-hoc shell formatting policy for migrated surfaces in this slice (hook rewrites, local hook command construction, PATH diagnostic IR projection, and shim text projection).
- [x] Windows runtime-specific behavior is explicit: PowerShell call-operator prefixes are emitted only for runtimes/shells that require them.
- [x] Path projection is shell-aware for migrated surfaces.
- [ ] `$HOME` / project-dir interpolation policy is fully centralized across all installer surfaces.
- [x] Existing regressions covered by this slice remain passing: `#2979`, `#3002`, `#3011`, `#3017`, `#3020`, `#3393`, `#3413`.
- [x] New tests assert against projection seam APIs (typed outputs) rather than installer source grep.
- [x] Internal subprocess execution remains array-arg / non-shell by default.
- [ ] ADR for the seam (`docs/adr/0009-shell-command-projection-module.md`) is included in this slice.

### Test coverage

Focused regression suites:
- `tests/bug-2979-hook-absolute-node.test.cjs`
- `tests/bug-3011-sdk-path-diagnostic.test.cjs`
- `tests/bug-2962-windows-sdk-shim.test.cjs`
- `tests/bug-3020-install-shell-path-probe.test.cjs`
- `tests/bug-3017-codex-hook-absolute-node.test.cjs`
- `tests/bug-3413-shell-command-projection.test.cjs`
- `tests/sh-hook-paths.test.cjs`

### Platforms tested

- [x] macOS
- [ ] Windows (including backslash path handling)
- [ ] Linux

### Runtimes tested

- [x] Claude Code
- [x] Gemini CLI
- [ ] OpenCode
- [x] Codex
- [ ] Copilot
- [x] Other: module-level projection tests (runtime-policy focused)

---

## Scope confirmation

- [x] The implementation matches the scope approved in the linked issue for this migration slice
- [x] No additional features, commands, or behaviors were added beyond this migration slice
- [ ] If scope changed during implementation, I updated the issue spec and received re-approval

---

## Checklist

- [x] Issue linked above with `Closes #NNN` — **PR will be auto-closed if missing**
- [x] Linked issue has the `approved-feature` label — **PR will be closed if missing**
- [x] All acceptance criteria from the issue are listed above with status
- [x] Implementation scope matches the approved spec slice
- [ ] All existing tests pass (`npm test`)
- [x] New tests cover the happy path, error cases, and edge cases
- [ ] `.changeset/` fragment added with a user-facing description of the feature (`npm run changeset -- --type Added --pr <NNN> --body "..."`)
- [ ] Documentation updated — commands, workflows, references, README if applicable
- [x] No unnecessary external dependencies added
- [x] Works on Windows (backslash paths handled)

## Breaking changes

None

## Summary by CodeRabbit

* **Bug Fixes**
  * Centralized shell/installer command projection so hook commands, Windows shims, and SDK PATH diagnostics behave consistently across platforms and runtimes.

* **Documentation**
  * Updated GitHub CLI guidance, added operational context and ADR for the projection module, and clarified installer migration semantics.

* **Tests**
  * Expanded and refactored tests to validate projection behavior across platforms and runtime variants.

[Review Change Stack](https://app.coderabbit.ai/change-stack/gsd-build/get-shit-done/pull/3445)

### CODE REVIEW

File: docs/installer-migrations.md [hunk 1]
```diff
@@ -360,7 +361,7 @@ for the new shape before changing migration behavior.
...
-| Antigravity | Skills in `skills/gsd-*/SKILL.md`; agents in `agents/`; Gemini-style `settings.json` hooks when installed by GSD | Global `ANTIGRAVITY_CONFIG_DIR` or `~/.gemini/antigravity`; local `./.agent` | GSD owns generated skills/agents/hooks and GSD settings entries only | Checked 2026-05-11 against available Antigravity install/config material; this row records GSD's Gemini-compatible settings contract as the compatibility baseline for installer migrations. |
+| Antigravity | Skills in `skills/gsd-*/SKILL.md`; agents in `agents/`; Gemini-style `settings.json` hooks when installed by GSD | Global `ANTIGRAVITY_CONFIG_DIR` or `~/.gemini/antigravity`; local `./.agent` | GSD owns generated skills/agents/hooks and GSD settings entries only | Public Antigravity install/config docs for this file layout were not stable or complete as of 2026-05-11; installer compatibility therefore uses GSD's Gemini-compatible settings contract as a documented shim baseline. |
```
[the remark this answers is not in the document]
Comment (trek-e) [reply docs/installer-migrations.md]: this is documentation, not code or context.
